components/script/dom/idbobjectstore.rs

```diff
@@ -0,0 +1,589 @@
+                                // FIXME:(rasviitanen) Handle this
+                                return;
+                            }
+
+                            if !desc.handle().obj.is_null() {
+                                *intermediate = desc.handle().value;
+                                has_prop = true;
+                            } else {
+                                // If we get here it means the object doesn't have the property or the
+                                // property is available throuch a getter. We don't want to call any
+                                // getters to avoid potential re-entrancy.
+                                // The blob object is special since its properties are available
+                                // only through getters but we still want to support them for key
+                                // extraction. So they need to be handled manually.
+                                unimplemented!("Blob tokens are not yet supported");
+                            }
+                        }
+
+                        if has_prop {
+                            // Treat undefined as an error
+                            if intermediate.is_undefined() {
+                                // FIXME:(rasviitanen) Throw/return error
+                                return;
+                            }
+
+                            if tokenizer.peek().is_some() {
+                                // ...and walk to it if there are more steps...
+                                *current_val = *intermediate;
+                            } else {
+                                // ...otherwise use it as key
+                                *return_val = *intermediate;
+                            }
+                        } else {
+                            *target_object = *object;
+                            target_object_prop_name = Some(token.to_string());
+                        }
+                    }
+
+                    if !target_object.get().is_null() {
+                        // We have started inserting new objects or are about to just insert
+                        // the first one.
+                        // FIXME:(rasviitanen) Implement this piece
+                        unimplemented!("keyPath tokens that requires insertion are not supported.");
+                    }
+                } // All tokens processed
+
+                if !target_object.get().is_null() {
+                    // If this fails, we lose, and the web page sees a magical property
+                    // appear on the object :-(
+                    unsafe {
+                        let prop_name_as_utf16: Vec<u16> =
+                            target_object_prop_name.unwrap().encode_utf16().collect();
+                        let succeeded: *mut ObjectOpResult =
+                            libc::malloc(mem::size_of::<ObjectOpResult>()) as *mut ObjectOpResult;
```

Comment:
Instead of malloc, use `let mut succeeded = ObjectOptResult { code_: ObjectOptResult_SpecialCodes::Uninitialized as usize };`